### PR TITLE
12 more tests for #661. Complete unit tests for Account controller...

### PR DESCRIPTION
**Pull request by klabranche**
Completed 12 more tests for #661 (write unit tests for Account controller). 
Finished out tests for reset password and forgot password functionality.  

Also fixed LogOffInvokesSignOutAsync & LogOffRedirectToCorrectActionAndController per https://github.com/HTBox/allReady/pull/747#issuecomment-220042509 by @mgmccarthy.   
Two tests were mocking out unneeded user manager object. 

This issue is almost complete after this PR.  Only 13 tests left for the External Login Callback functionality. :-)

### CODE REVIEW

**Comment by mgmccarthy** [comment]
@klabranche, I'll be able to get to a code review on these unit tests tonight. Thanks!

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AccountControllerTests.cs [hunk 1]
```diff
@@ -802,39 +800,156 @@ public void ForgotPasswordGetHasAllowAnonymousAttribute()
...
+
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper.Setup(x => x.Action(It.IsAny<UrlActionContext>())).Returns(callbackUrl);
+
+            sut.SetFakeHttpRequestSchemeTo(It.IsAny<string>());
+            sut.Url = urlHelper.Object;
+
+            await sut.ForgotPassword(vm);
+
+            emailSender.Verify(x => x.SendEmailAsync(email, It.IsAny<string>(), It.Is<string>(y => y.Contains(callbackUrl))), Times.Once);
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AccountControllerTests.cs]: we want to be explicit with this test for all the values passed to `SendEmailAsync`. Right now, we're just testing that the third parameter contains the `callbackUrl`. I'd re-write as this:
`emailSender.Verify(x => x.SendEmailAsync(email, "Reset allReady Password", $"Please reset your allReady password by clicking here: <a href=\"{callbackUrl}\">link</a>"), Times.Once);`

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AccountControllerTests.cs [hunk 2]
```diff
@@ -896,53 +1011,160 @@ public void ResetPasswordGetHasAllowAnonymousAttribute()
...
+
+            var userManager = CreateUserManagerMock();
+            var identityResult = IdentityResult.Failed(new IdentityError { Description = "IdentityErrorDescription" });
+            var user = new ApplicationUser();
+
+            userManager.Setup(x => x.FindByNameAsync(email)).Returns(() => Task.FromResult(user));
+            userManager.Setup(x => x.ResetPasswordAsync(user, It.IsAny<string>(), It.IsAny<string>())).Returns(() => Task.FromResult(identityResult));
+            var sut = new AccountController(userManager.Object, null, null, null, null);
+
+            var result = await sut.ResetPassword(vm) as ViewResult;
```
Comment (mgmccarthy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/AccountControllerTests.cs]: you don't need `var sut` here

**Comment by mgmccarthy** [comment]
@klabranche, just a couple quick comments that you can fix if you have time. @MisterJames @tonysurma, please accept this PR. @klabranche, thanks for the contribution!

**Comment by tonysurma** [comment]
@mgmccarthy so accepting as is, @klabranche thanks and thanks in advance if you make any more contributions!

**Pull request merged**